### ISSUE TITLE
VMware ovf file contains unsupported value of disk format

### ISSUE THREAD

**Issue by muchengshu** [ilbers/isar#106]
Problem Description: 
When I import an ovf file generated by Isar into ESXi, I found there was an error message: Not supported value "[URL]" for attribute "format" of element "disk". 
 
Expected behavior: 
The ovf file should contain valid value of disk format: e.g. http://www.vmware.com/interfaces/specifications/vmdk.html#streamOptimized. 
 
Actual behavior: 
The ovf file contains invalid value of disk format: http://www.vmware.com/interfaces/specifications/vmdk.html#. 
 
How to reproduce: 
Build the VMware image with Isar and unpack the generated ova file, the ovf file could be readable.

**Comment by ismagulb** [comment]
Thanks Shu Mucheng (please correct) for the report,
which specific value does Isar generate?
Which value would you like to have instead?

**Comment by WiseLord** [comment]
@muchengshu could you clarify where the errors happens?

I tried to build virtualbox image and imported it with no errors:

```
isar $ vboxmanage import build/tmp/deploy/images/virtualbox/isar-image-base-debian-bookworm-virtualbox.ovf 
0%...10%...20%...30%...40%...50%...60%...70%...80%...90%...100%
Interpreting /home/Work/isar/build/tmp/deploy/images/virtualbox/isar-image-base-debian-bookworm-virtualbox.ovf...
OK.
Disks:
  vmdisk1       1680332800      -1      http://www.vmware.com/interfaces/specifications/vmdk.html#      isar-image-base-debian-bookworm-virtualbox-disk001.vmdk -1      -1

Virtual system 0:
 0: Suggested OS type: "Debian_64"
    (change with "--vsys 0 --ostype <type>"; use "list ostypes" to list all possible values)
 1: Suggested VM name "isar-image-base-debian-bookworm-virtualbox 6"
    (change with "--vsys 0 --vmname <name>")
 2: Suggested VM group "/"
    (change with "--vsys 0 --group <group>")
 3: Suggested VM settings file name "/home/Share/VirtualBox/isar-image-base-debian-bookworm-virtualbox 6/isar-image-base-debian-bookworm-virtualbox 6.vbox"
    (change with "--vsys 0 --settingsfile <filename>")
 4: Suggested VM base folder "/home/Share/VirtualBox"
    (change with "--vsys 0 --basefolder <path>")
 5: Number of CPUs: 4
    (change with "--vsys 0 --cpus <n>")
 6: Guest memory: 8192 MB
    (change with "--vsys 0 --memory <MB>")
 7: Sound card (appliance expects "ensoniq1371", can change on import)
    (disable with "--vsys 0 --unit 7 --ignore")
 8: USB controller
    (disable with "--vsys 0 --unit 8 --ignore")
 9: Network adapter: orig NAT, config 6, extra slot=0;type=NAT
10: CD-ROM
    (disable with "--vsys 0 --unit 10 --ignore")
11: IDE controller, type PIIX4
    (disable with "--vsys 0 --unit 11 --ignore")
12: IDE controller, type PIIX4
    (disable with "--vsys 0 --unit 12 --ignore")
13: VirtioSCSI controller, type VirtioSCSI
    (disable with "--vsys 0 --unit 13 --ignore")
14: Hard disk image: source image=isar-image-base-debian-bookworm-virtualbox-disk001.vmdk, target path=isar-image-base-debian-bookworm-virtualbox-disk001.vmdk, controller=13;port=0
    (change target path with "--vsys 0 --unit 14 --disk path";
    change controller with "--vsys 0 --unit 14 --controller <index>";
    change controller port with "--vsys 0 --unit 14 --port <n>";
    disable with "--vsys 0 --unit 14 --ignore")
0%...10%...20%...30%...40%...50%...60%...70%...80%...90%...100%
Successfully imported the appliance.
```

Imported machine boots OK.

**Comment by muchengshu** [comment]
> Thanks Shu Mucheng (please correct) for the report, which specific value does Isar generate? Which value would you like to have instead?

When I build the ova image, meta-isar/conf/machine/vmware.conf will specify the OVF_TEMPLATE_FILE to meta/recipes-core/images/vmware/vm-img-vmware.ovf.tmpl. After investigation, I found the variable VMDK_SUBFORMAT was not assigned to the template file(meta/classes/imagetypes_vm.bbclass, line 63). I have posted the patch in mail list isar-users, please check the mail for details, thanks.

**Comment by muchengshu** [comment]
> @muchengshu could you clarify where the error happens?
> 
> I tried to build virtualbox image and imported it with no errors:
> 
> ```
> isar $ vboxmanage import build/tmp/deploy/images/virtualbox/isar-image-base-debian-bookworm-virtualbox.ovf 
> 0%...10%...20%...30%...40%...50%...60%...70%...80%...90%...100%
> Interpreting /home/Work/isar/build/tmp/deploy/images/virtualbox/isar-image-base-debian-bookworm-virtualbox.ovf...
> OK.
> Disks:
>   vmdisk1       1680332800      -1      http://www.vmware.com/interfaces/specifications/vmdk.html#      isar-image-base-debian-bookworm-virtualbox-disk001.vmdk -1      -1
> 
> Virtual system 0:
>  0: Suggested OS type: "Debian_64"
>     (change with "--vsys 0 --ostype <type>"; use "list ostypes" to list all possible values)
>  1: Suggested VM name "isar-image-base-debian-bookworm-virtualbox 6"
>     (change with "--vsys 0 --vmname <name>")
>  2: Suggested VM group "/"
>     (change with "--vsys 0 --group <group>")
>  3: Suggested VM settings file name "/home/Share/VirtualBox/isar-image-base-debian-bookworm-virtualbox 6/isar-image-base-debian-bookworm-virtualbox 6.vbox"
>     (change with "--vsys 0 --settingsfile <filename>")
>  4: Suggested VM base folder "/home/Share/VirtualBox"
>     (change with "--vsys 0 --basefolder <path>")
>  5: Number of CPUs: 4
>     (change with "--vsys 0 --cpus <n>")
>  6: Guest memory: 8192 MB
>     (change with "--vsys 0 --memory <MB>")
>  7: Sound card (appliance expects "ensoniq1371", can change on import)
>     (disable with "--vsys 0 --unit 7 --ignore")
>  8: USB controller
>     (disable with "--vsys 0 --unit 8 --ignore")
>  9: Network adapter: orig NAT, config 6, extra slot=0;type=NAT
> 10: CD-ROM
>     (disable with "--vsys 0 --unit 10 --ignore")
> 11: IDE controller, type PIIX4
>     (disable with "--vsys 0 --unit 11 --ignore")
> 12: IDE controller, type PIIX4
>     (disable with "--vsys 0 --unit 12 --ignore")
> 13: VirtioSCSI controller, type VirtioSCSI
>     (disable with "--vsys 0 --unit 13 --ignore")
> 14: Hard disk image: source image=isar-image-base-debian-bookworm-virtualbox-disk001.vmdk, target path=isar-image-base-debian-bookworm-virtualbox-disk001.vmdk, controller=13;port=0
>     (change target path with "--vsys 0 --unit 14 --disk path";
>     change controller with "--vsys 0 --unit 14 --controller <index>";
>     change controller port with "--vsys 0 --unit 14 --port <n>";
>     disable with "--vsys 0 --unit 14 --ignore")
> 0%...10%...20%...30%...40%...50%...60%...70%...80%...90%...100%
> Successfully imported the appliance.
> ```
> 
> Imported machine boots OK.

Thanks for your reply. VirtualBox and VMware workstation are both OK for that. As I mentioned, it just comes out in ESXi(an OS installed on servers from VMware).

**Comment by WiseLord** [comment]
Yes, using VMDK_SUBFORMAT inside .tmpl files without adding them to OVA_VARS looks incorrect.
